frontend/src/main/java/org/abs_models/frontend/typechecker/ext/TypeSystemExtension.java

```diff
@@ -22,6 +22,8 @@
 
     void checkAssignable(Type adaptTo, AdaptDirection dir, Type rht, Type lht, ASTNode<?> n);
 
+    void checkAssignable(Type l, Exp r, ASTNode<?> n);
```

Comment:
See comment in `TypeExtensionHelper.java`.  Can these two methods be merged, or renamed so the reader sees what each one does?